src/webpack.js

```diff
@@ -29,13 +32,131 @@ const parseEntrypoint = (entrypoint) => {
   };
 };
 
+/**
+ * Helper function to trim absolute file path by removing the `process.cwd()`
+ * prefix if file is nested under `process.cwd()`.
+ */
+function makeFilePathRelativeToCwd (file) {
+  const cwd = process.cwd();
+  return (file.startsWith(cwd)) ? '.' + file.substring(cwd.length) : file;
+}
+
+/**
+ * @param {String} outputDir the directory that contains output files
+ * @param {String[]} entryNames the entrypoint names from which output was produced
+ */
+async function zipOutputFiles (outputDir, entryNames) {
+  console.log('\nCreating zip file for each entrypoint...\n');
+
+  for (const entryName of entryNames) {
+    const dirname = path.dirname(entryName);
+    // Find all of the output files that belong to this entry
+    const entriesForZipFile = (await glob(`${entryName}*`, {
+      cwd: outputDir
+    })).map((file) => {
+      return {
+        name: (dirname === '.') ? file : file.substring(dirname.length + 1),
+        file: path.join(outputDir, file)
+      };
+    });
+
+    // Now, write a zip file for each entry
+    console.log(`Creating zip for entrypoint ${chalk.bold(entryName)}...`);
+    console.log(entriesForZipFile.map((entry) => {
+      return `- ${chalk.bold(entry.name)}\n`;
+    }).join(''));
+    const outputZipFile = path.join(outputDir, `${entryName}.zip`);
+    await zip(outputZipFile, entriesForZipFile);
+    console.log(chalk.green(`Zip file for ${chalk.bold(entryName)} written to ` +
+      `${chalk.bold(makeFilePathRelativeToCwd(outputZipFile))}\n`));
+  }
+}
+
+const INDEX_FILES = ['index.js', 'index.ts'];
+
+async function findIndexFile (dir) {
+  for (const indexFile of INDEX_FILES) {
+    const candidateFile = path.join(dir, indexFile);
+    try {
+      const stats = await fs.stat(candidateFile);
+      if (!stats.isDirectory()) {
+        return candidateFile;
+      }
+    } catch (ignoreErr) {
+      // ignore error
```

Comment:
Is it really safe to ignore the error here? This seems like it could hide real problems from the consumer like lack of access to the directory.

Reply:
I think the `catch` with ignore is enough... the developer can dig deeper and look at the warning if no index file is found to further investigatee.

Reply:
Unless I misread the code, there is no warning. Rather the entry is silently ignored. I also disagree that a generic warning is sufficient. The only error that seems safe to ignore is if the file doesn't exist. Real I/O errors should be reported to the user.

Reply:
Found the error right below. Still we need details as to why. It would be confusing to see "no index file was found" and see an index file in the directory.

Reply:
I added a check for `err.code !== 'ENOENT'` and log warning with actual error message if we fail to stat a candidate index file.

Reply:
Cool, works for me. Thanks!